src/connectors/breadcrumb/connectBreadcrumb.js

```diff
@@ -0,0 +1,194 @@
+              name: hierarchicalFacetName,
+              separator,
+              rootPath,
+            },
+          ],
+        };
+      },
+
+      init({ createURL, helper, instantSearchInstance }) {
+        this._createURL = facetValue => {
+          if (!facetValue) {
+            const breadcrumb = helper.getHierarchicalFacetBreadcrumb(
+              hierarchicalFacetName
+            );
+            if (breadcrumb.length > 0) {
+              return createURL(
+                helper.state.toggleRefinement(
+                  hierarchicalFacetName,
+                  breadcrumb[0]
+                )
+              );
+            } else {
+              return undefined;
+            }
+          }
+          return createURL(
+            helper.state.toggleRefinement(hierarchicalFacetName, facetValue)
+          );
+        };
+
+        this._refine = function(facetValue) {
+          if (!facetValue) {
+            const breadcrumb = helper.getHierarchicalFacetBreadcrumb(
+              hierarchicalFacetName
+            );
+            if (breadcrumb.length > 0) {
+              helper
+                .toggleRefinement(hierarchicalFacetName, breadcrumb[0])
+                .search();
+            }
+          } else {
+            helper.toggleRefinement(hierarchicalFacetName, facetValue).search();
+          }
+        };
+
+        renderFn(
+          {
+            createURL: this._createURL,
+            canRefine: false,
+            instantSearchInstance,
+            items: [],
+            refine: this._refine,
+            widgetParams,
+          },
+          true
+        );
+      },
+
+      render({ instantSearchInstance, results, state }) {
+        if (
+          !state.hierarchicalFacets ||
+          (Array.isArray(state.hierarchicalFacets) &&
+            state.hierarchicalFacets.length === 0)
+        ) {
+          throw new Error(usage);
+        }
+
+        const [{ name: facetName }] = state.hierarchicalFacets;
+
+        const facetsValues = results.getFacetValues(facetName);
+        const items = processArray(prepareItems(facetsValues));
+
+        renderFn(
+          {
+            canRefine: items.length > 0,
+            createURL: this._createURL,
+            instantSearchInstance,
+            items,
+            refine: this._refine,
+            widgetParams,
+          },
+          false
+        );
+      },
+    };
+  };
+}
+
+function prepareItems(obj) {
+  return obj.data.reduce((result, currentItem) => {
+    if (currentItem.isRefined) {
+      result.push({
+        name: currentItem.name,
+        value: currentItem.path,
+      });
+      if (Array.isArray(currentItem.data)) {
+        const children = prepareItems(currentItem);
+        result = result.concat(children);
+      }
+    }
+    return result;
+  }, []);
+}
+
+function processArray(array) {
+  return array.map((x, idx) => ({
+    name: x.name,
+    value: idx + 1 == array.length ? null : array[idx + 1].value,
```

Comment:
We should only use `==` when we want JS to make [type coercion](https://code.tutsplus.com/articles/the-beginners-guide-to-type-coercion-a-practical-example--cms-21998) automatically. Since we're comparing numbers, it should not be an issue to use `===`.